### DISCUSSION TITLE
Custom validation for file uploads

### DISCUSSION THREAD

**Discussion by tanthammar** [livewire/livewire#1475] [4. General]
The current way to validate file uploads are based on a single setting in the config file.
It is rarely the case that file upload fields in all forms share the same rules.

One form may accept Excel and another images, and so on.

The current setup gives unwanted payload and storage.
One has to write custom validators to avoid the problem, but it is impossible to avoid unwanted file uploads.

May I suggest that the the validation rules are collected from the `$rules[$key] ?? config()` instead.

**Comment by nuernbergerA** [comment]
I was in the same situation and decided to just use [accept](https://www.geeksforgeeks.org/html-input-accept-attribute/) attribute on the `input` itself and using validation rules.

I know the files could technically make it to the `tmp` folder - but will be deleted after 24h, which is not a big deal for me
